rocketmq-spring-boot/src/test/java/org/apache/rocketmq/spring/metric/MetricExtensionTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.rocketmq.spring.metric;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class MetricExtensionTest implements MetricExtension {
+
+    private static String TOPIC;
+    private static int COUNT;
+    private static EConsumerMode CONSUMER_MODE;
+
+    @Override
+    public void addProducerMessageCount(String topic, int count) {
+        TOPIC = topic;
+        COUNT = count;
+    }
+
+    @Override
+    public void addConsumerMessageCount(String topic, int count, EConsumerMode consumerMode) {
+        TOPIC = topic;
+        COUNT = count;
+        CONSUMER_MODE = consumerMode;
+    }
+
+    @Test
+    public void testAddProducerMessageCount() {
+        MetricExtensionProvider.addProducerMessageCount("topic1", 111);
```

Comment:
It could be helpful for your code convincible if you could make more corner cases, such as have you simulated the count of failures?

Reply:
This unit test is mainly used to test whether the SPI of MetricExtensionProvider is running normally.The scenario where the counting fails you mentioned was considered by the developer when implementing the MetricExtension interface.